app/src/lib/dispatcher/app-store.ts

```diff
@@ -1507,28 +1507,28 @@ export class AppStore {
   }
 
   /** This shouldn't be called directly. See `Dispatcher`. */
-  public _deleteBranch(repository: Repository, branch: Branch): Promise<void> {
-    return this.withAuthenticatingUser(
```

Comment:
Now that we pass the accounts to this method, we can defer authentication to the methods that exist in `lib/git/`. Like the case [here](https://github.com/desktop/desktop/blob/master/app/src/lib/git/branch.ts#L76)